my_methods.rb

```diff
@@ -0,0 +1,199 @@
+# rubocop: disable Metrics/ModuleLength
+# rubocop: disable Metrics/MethodLength
+# rubocop: disable Style/For
+# rubocop: disable Metrics/BlockNesting
+# rubocop: disable Metrics/CyclomaticComplexity
+# rubocop: disable Metrics/PerceivedComplexity
+# rubocop: disable Style/IfInsideElse
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    array = self
+    array = array.to_a if array.class == Range
+    array = array.values if array.class == Hash
+
+    for item in array
+      yield item
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    array = self
+    array = array.to_a if array.class == Range
+    array = array.values if array.class == Hash
+
+    for i in (0...array.length)
+      yield array[i], i
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    array = self
+    array = array.to_a if array.class == Range
+    array = array.values if array.class == Hash
+
+    res = []
+
+    for i in array
+      res.push(i) if yield i
+    end
+    res
+  end
+
+  def my_all?(arg = nil)
+    return false if !block_given? && !arg
+
+    array = self
+    array = array.to_a if array.class == Range
+    array = array.values if array.class == Hash
+
+    return true if array.empty?
+
+    for i in 0...array.length
+      if arg
+        if arg.is_a? Module or arg.is_a? Class
+          return false unless array[i].is_a?(arg)
+        elsif arg.class == Regexp
+          return false if array[i].match(arg).nil?
+        else return false unless array[i] != arg
+        end
+      else return false unless yield array[i]
+      end
+    end
+
+    true
+  end
+
+  def my_any?(arg = nil)
+    array = self
+    array = array.to_a if array.class == Range
+    array = array.values if array.class == Hash
+
+    return false if array.empty?
+    return true if array.include? true
+
+    for i in 0...array.length
+      if arg
+        if arg.is_a? Module or arg.is_a? Class
+          return true if array[i].is_a?(arg)
+        elsif arg.class == Regexp
+          return true if array[i].match(arg).nil?
+        else return true unless array[i] != arg
+        end
+      else return true if yield array[i]
+      end
+    end
+    false
+  end
+
+  def my_none?(arg = nil)
+    return true if !block_given? && !arg
+
+    array = self
+    array = array.to_a if array.class == Range
+    array = array.values if array.class == Hash
+
+    for i in 0...array.length
+      if arg
+        if arg.is_a? Module or arg.is_a? Class
+          return false if array[i].is_a?(arg)
+        elsif arg.class == Regexp
+          return false if array[i].match(arg).nil?
+        else return false unless array[i] != arg
+        end
+      else return false if yield array[i]
+      end
+    end
+
+    true
+  end
```

Comment:
Fix the following for `my_none?` method

```
# It should return true since none of the words contain letter d.
puts %w{ant bear cat}.none?(/d/)    #=> true
puts %w{ant bear cat}.my_none?(/d/)  #=> false

puts [nil, false, true].none?          #=> false
puts [nil, false, true].my_none?   #=> true

puts (0..4).none?    #=> false
puts (0..4).my_none?  #=> true
```